packages/wallet/api/src/lib-wallet.js

```diff
@@ -556,6 +556,46 @@ export function makeWallet({
       });
   }
 
+  /**
+   * @param { () => Promise } get - The function whose return value to memoize.
+   */
+  const makeMemoizedGetter = get => {
+    let pk;
+    const doGet = async () => {
+      const data = await get();
+      pk.resolve(data);
+    };
+
+    return () => {
+      if (pk === undefined) {
+        pk = makePromiseKit();
+        doGet();
+      }
+      return pk.promise;
+    };
+  };
+
+  const getAttIssuer = makeMemoizedGetter(async () =>
```

Comment:
Most `async` functions here can be synchronous functions returning `E()`

Reply:
Done